cli/src/commands/extensions/api.rs

```diff
@@ -0,0 +1,175 @@
+//! intentional, as there are no clients for those functions in the rest of the code, but
+//! those functions will have to be used by the Deno integration; at that point, we may
+//! remove the annotations.
+
+use std::path::Path;
+use std::str::FromStr;
+
+use crate::commands::parse::{get_packages_from_lockfile, LOCKFILE_PARSERS};
+use crate::config::{get_current_project, Config};
+use crate::{api::PhylumApi, auth::UserInfo};
+
+use anyhow::{anyhow, Context, Error, Result};
+use deno_core::OpState;
+use once_cell::sync::Lazy;
+use phylum_types::types::auth::{AccessToken, RefreshToken};
+use phylum_types::types::common::{JobId, ProjectId};
+use phylum_types::types::job::JobStatusResponse;
+use phylum_types::types::package::{
+    Package, PackageDescriptor, PackageStatusExtended, PackageType,
+};
+use phylum_types::types::project::ProjectDetailsResponse;
+
+/// Container for lazily evaluated dependencies of Extensions API functions. These values won't be
+/// visible from extension code and will have to be set up by the JS runtime builder which will
+/// load their configuration as any other command and then provide the pertinent factories.
+struct InjectedDependencies {
+    api: Lazy<PhylumApi>,
+    config: Lazy<Config>,
+}
+
+impl InjectedDependencies {
+    #[allow(unused)]
+    pub(super) fn from_factories(
+        api_factory: fn() -> PhylumApi,
+        config_factory: fn() -> Config,
+    ) -> Self {
+        InjectedDependencies {
+            api: Lazy::new(api_factory),
+            config: Lazy::new(config_factory),
+        }
+    }
+}
+
+#[allow(unused)]
+pub(super) async fn phylum_analyze(
+    state: &mut OpState,
+    lockfile: &str,
+    project: Option<&str>,
+    group: Option<&str>,
+) -> Result<ProjectId> {
+    let api = &mut state.borrow_mut::<InjectedDependencies>().api;
+    let (packages, request_type) = get_packages_from_lockfile(Path::new(lockfile))
+        .context("Unable to locate any valid package in package lockfile")?;
+
+    let (project, group) = match (project, group) {
+        (Some(project), group) => (api.get_project_id(project, group).await?, None),
+        (None, _) => {
+            if let Some(p) = get_current_project() {
+                (p.id, p.group_name)
+            } else {
+                return Err(anyhow!("Failed to find a valid project configuration"));
+            }
+        }
+    };
+
+    let job_id = api
+        .submit_request(
+            &request_type,
+            &packages,
+            false,
+            project,
+            None,
+            group.map(String::from),
+        )
+        .await?;
+
+    Ok(job_id)
+}
+
+#[allow(unused)]
+pub(super) async fn phylum_auth_status(state: &mut OpState) -> Result<UserInfo> {
+    let deps = state.borrow_mut::<InjectedDependencies>();
+    deps.api
+        .user_info(&deps.config.auth_info)
+        .await
+        .map_err(Error::from)
+}
+
+#[allow(unused)]
+pub(super) async fn phylum_auth_token_bearer(
```

Comment:
Same reasoning as above; I'm totally ok with a different naming convention, but in that case we need to document the correspondence between CLI endpoints and extensions API.

Reply:
There's a big difference between `phylum auth token --bearer` and `phylum_auth_token_bearer`. I don't think it's appropriate to mimic the literal words of the CLI, since CLIs are a syntax with certain meaning. Instead of translating that word-for-word, we should translate its meaning.

So for the example `phylum auth token --bearer`, we're clearly not trying to get `auth token bearer` we're telling the CLI, that we want the token of the type bearer.

Reply:
That makes sense, but we still do want to retain correspondence with CLI endpoints. I will add doc comments that say "This is the equivalent of `phylum foo --bar`".

Reply:
Agreed, documentation is probably the best way to go.